### PR TITLE
feat(dashboard): Product create from - details

### PR DESCRIPTION
**Pull request by kasperkristensen**
**What**
- First part of the product creation form.
- New components:
  - ChipInput - Allows users to input chips into a input field. Chips are created by hitting the `,` or `Enter / Return` keys. Deleting a chip is done by hitting `Backspace` when the cursor is next to chip, or clicking the `X` button in the chip. Used for inputting option values.
  - SortableList - A sortable drag-n-drop list that allows the user to re-arrange the order of items. Used for re-arranging the ranking of variants.
  - ChipGroup - New re-usable component that is used to render a group of values as Chips. This should be used for SplitView form items.
  - CategoryCombobox - (WIP) Nested Combobox component for selecting multiple categories a product should be associated with.
- New hooks:
  - useComboboxData - Hook for easily managing the state of comboboxes.
  - useDebouncedSearch - Hook for managing debounced search queries.

### CODE REVIEW

File: packages/admin-next/dashboard/src/components/inputs/chip-input/chip-input.tsx [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+  onBlur?: () => void
+  name?: string
+  disabled?: boolean
+  allowDuplicates?: boolean
+  showRemove?: boolean
+  variant?: "base" | "contrast"
+  className?: string
+}
+
+export const ChipInput = forwardRef<HTMLInputElement, ChipInputProps>(
```
Comment (fPolic) [inline packages/admin-next/dashboard/src/components/inputs/chip-input/chip-input.tsx:25]: nit: can we create a last chip on blur (or at least remove the input that hasn't been converted to a chip)?

File: packages/admin-next/dashboard/src/v2-routes/products/common/components/category-combobox/category-combobox.tsx [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+  const options = getOptions(product_categories || [])
+
+  const showTag = value.length > 0 && !open
+
+  if (isError) {
+    throw error
+  }
+
+  return (
+    <Popover.Root open={open} onOpenChange={handleOpenChange}>
```
Comment (fPolic) [inline packages/admin-next/dashboard/src/v2-routes/products/common/components/category-combobox/category-combobox.tsx:159]: nit: when this component is near the bottom of the screen, when you type, the popover sometimes renders above the input while it loads and then is rendered below the input when the fetch is over. Can we force the popover to always render to the bottom of the input?

File: packages/admin-next/dashboard/src/v2-routes/products/product-create/components/product-create-details-form/components/product-create-details-variant-section/product-create-details-variant-section.tsx [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+
+                                return (
+                                  <Form.Item className="flex flex-col space-y-1.5">
+                                    <div className="flex items-center pl-2">
+                                      <Form.Label className="txt-compact-xsmall-plus text-ui-fg-muted">
+                                        {t("fields.values")}
+                                      </Form.Label>
+                                    </div>
+                                    <Form.Control>
+                                      <ChipInput
```
Comment (fPolic) [inline packages/admin-next/dashboard/src/v2-routes/products/product-create/components/product-create-details-form/components/product-create-details-variant-section/product-create-details-variant-section.tsx]: praise: love this

**Review by fPolic — APPROVED** [review]
Great work @kasperkristensen 🔥

File: packages/admin-next/dashboard/src/v2-routes/products/common/components/category-combobox/category-combobox.tsx (hunk 2 above)
Comment (kasperkristensen) [reply packages/admin-next/dashboard/src/v2-routes/products/common/components/category-combobox/category-combobox.tsx:159]: I think its better to let it decide where it has the space to render properly, but the component is still missing an "official" design so maybe we can come up with a smaller loading state skeleton, so it does not flicker so much when there are few options in the result.

**Pull request merged**